### ISSUE TITLE
Submission address != POP3 username

### ISSUE THREAD

**Issue by flamble-greg** [arnoldle/phplist-plugin-submitByMailPlugin#8]
Version: 1.0b2.12

Please add a field to allow a POP3 username to be specified. Currently, the username is assumed to be the submission address, which is not the case for some mail servers.

**Comment by arnoldle** [comment]
Do you have some documentation for this? I put the system together with hosting services in mind. The services with which I have experience all require the entire address as the user name, for example, ‘auser@nowhere.org’.

If you know of systems that do it differently, I would like to have the information. 

Arnold

**Comment by flamble-greg** [comment]
Hi Arnold

Certainly. Take Dovecot (one of the most popular open source POP3/IMAP servers). By default, it utilises local user accounts for authentication: http://wiki2.dovecot.org/PasswordDatabase

Although the submission address is phplist@example.com, the username might simply be 'phplist'. While I accept that there are way to work around this at a server level, it would be a useful feature to have to be able to specify a separate POP3 username to be used by submitByMail.

Greg

**Comment by arnoldle** [comment]
I think that I can fix this, but there are implications for most of the code files. It will take a few days.

Arnold

**Comment by arnoldle** [comment]
Hi Greg!

I have modified the plugin so that it tries both the submission address and the user name (generated from the submission address by taking the part of the address before the ‘@‘) as the POP3 login. When a list is configured, it tries the submission address as the login. If that does not work, it tries the user name. It remembers which one works, and then uses that one to download mail for the list.

My problem is that I cannot find a server using only the user name. I have looked among the free email services, as well as the paid hosting services that I have access to. None use just the user name.

So I cannot test this new feature. Since you asked for this, perhaps you might test this feature for me. If you send me your email address, I will send you a zip file of the newest version. I don’t want to do a public release until the version is tested.

Arnold
